samples/client/echo_api/python/tests/test_manual.py

```diff
@@ -174,15 +174,12 @@ def testBodyParameter(self):
         n = openapi_client.Pet.from_dict({"name": "testing", "photoUrls": ["http://1", "http://2"]})
         api_instance = openapi_client.BodyApi()
         api_response = api_instance.test_echo_body_pet_response_string(n)
-        self.assertEqual(api_response, '{"name": "testing", "photoUrls": ["http://1", "http://2"]}')
+        self.assertEqual(api_response, "{'name': 'testing', 'photoUrls': ['http://1', 'http://2']}")
 
         t = openapi_client.Tag()
         api_response = api_instance.test_echo_body_tag_response_string(t)
         self.assertEqual(api_response, "{}") # assertion to ensure {} is sent in the body
 
-        api_response = api_instance.test_echo_body_tag_response_string(None)
-        self.assertEqual(api_response, "") # assertion to ensure emtpy string is sent in the body
```

Comment:
This fails because None is not a json.
This test is erroneous and has been removed.

Reply:
the test is for optional body parameter (e.g. Pet). shall we keep it instead?

Reply:
> the test is for optional body parameter (e.g. Pet).

If that's the case, this test should be performed in `petstore_api` instead of `echo_api`.

Since an empty string is not in JSON format, it conflicts with the `Content-Type: application/json` in the response header.
Alternatively, is there an option to change the response header of echo_api to `Content-Type: text/plain` ?

Reply:
Using Pet just an example. It can be any models (which will be serialized to JSON string before sending to the server).

> Since an empty string is not in JSON format, it conflicts with the Content-Type: application/json in the response header.

Understood as confirmed in https://github.com/axios/axios/issues/4146 but please take a look at https://github.com/axios/axios/issues/4146 as well, which is also a "use case" presented to use previously (and that's why we've added a test to cover it)

With this change, if None is provided, what will be sent to the server? just empty JSON string `""` ?

Reply:
> Using Pet just an example. It can be any models (which will be serialized to JSON string before sending to the server).

In this case, `None` is not serialized to `Pet`. Therefore, it is not serialized into a JSON string before being sent to the server.  Only empty json will be sent.


> but please take a look at https://github.com/axios/axios/issues/4146 as well, which is also a "use case" presented to use previously (and that's why we've added a test to cover it)

Ok, add a process to prevent json deserialize if the response is empty.

> With this change, if None is provided, what will be sent to the server? just empty JSON string "" ?

This change does not change what is sent to the server.  An empty json is sent.


Currently, empty JSON is acceptable when the client sends a request.
You are correct, receiving should also be allowed.

Reply:
Allow empty json in response
https://github.com/OpenAPITools/openapi-generator/pull/18800/commits/e77545e7aa0e28ce6cf383c22e58b9ecf9850434#diff-c1d7ea3a1de60effaa5908cee1be9a7e2fd420bdb75a2e2a28a6c3290380192dR411-R414